be/src/exec/es/es_scroll_parser.cpp

```diff
@@ -519,7 +521,178 @@ Status ScrollParser::fill_columns(const TupleDescriptor* tuple_desc,
+                        if (sub_col[0].IsString()) {
+                            array.push_back(get_datev2_int(sub_col, true));
+                            break;
+                        }
+                        // ES would return millisecond timestamp for date field, divided by 1000 because the unit for from_unixtime is seconds
+                        array.push_back(get_datev2_int(sub_col, false));
+                    } else {
+                        // this would happened just only when `enable_docvalue_scan = false`, and field has string format date from _source
+                        RETURN_ERROR_IF_COL_IS_ARRAY(sub_col, type);
+                        RETURN_ERROR_IF_COL_IS_NOT_STRING(sub_col, type);
+                        array.push_back(get_datev2_int(sub_col, true));
+                    }
+                    break;
+                }
+                case TYPE_DATETIMEV2: {
+                    // this would happend just only when `enable_docvalue_scan = false`, and field has timestamp format date from _source
+                    if (sub_col.IsNumber()) {
+                        // ES process date/datetime field would use millisecond timestamp for index or docvalue
+                        // processing date type field, if a number is encountered, Doris On ES will force it to be processed according to ms
+                        // Doris On ES needs to be consistent with ES, so just divided by 1000 because the unit for from_unixtime is seconds
+                        array.push_back(get_datetimev2_int(sub_col, false));
+                    } else if (sub_col.IsArray() && pure_doc_value) {
+                        // this would happened just only when `enable_docvalue_scan = true`
+                        // ES add default format for all field after ES 6.4, if we not provided format for `date` field ES would impose
+                        // a standard date-format for date field as `2020-06-16T00:00:00.000Z`
+                        // At present, we just process this string format date. After some PR were merged into Doris, we would impose `epoch_mills` for
+                        // date field's docvalue
+                        if (sub_col[0].IsString()) {
+                            array.push_back(get_datetimev2_int(sub_col, true));
+                            break;
+                        }
+                        // ES would return millisecond timestamp for date field, divided by 1000 because the unit for from_unixtime is seconds
+                        array.push_back(get_datetimev2_int(sub_col, false));
+                    } else {
+                        // this would happened just only when `enable_docvalue_scan = false`, and field has string format date from _source
+                        RETURN_ERROR_IF_COL_IS_ARRAY(sub_col, type);
+                        RETURN_ERROR_IF_COL_IS_NOT_STRING(sub_col, type);
+                        array.push_back(get_datetimev2_int(sub_col, true));
```

Comment:
As discussion with @stalary, it's better to place `DateV2` and `DateTimeV2` in one method since these case branch code were too similar.